### ISSUE TITLE
Reopening a credit note paid with status unpaid

### ISSUE THREAD

**Issue by Cirillo007** [Dolibarr/dolibarr#32350]
### Bug

Having a credit note paid (supplier invoice), we reopened the credit note, but the status became unpaid.

This credit note is 'PAID' because we used the credit available for the supplier.
The 120 euros will be used later as discount with the next invoice.

So here is the credit note invoice (supplier)
<img width="1021" alt="image" src="https://github.com/user-attachments/assets/b94dafb5-1a38-4bd5-9229-e494d896f32f" />

If now we reopen the credit note using 'ROUVRIR' it becomes:
<img width="1023" alt="image" src="https://github.com/user-attachments/assets/3f8ddfbd-c8b4-444e-9029-ab4d3e59632e" />

Then the credit becomes becomes unpaid (with the credit available).
Two things:
1/ is it remains unpaid, the credit available for later invoices has to be cancelled
2/ the credit note invoice must remains paid

Because we can have a trouble (double use) if we make a payment on it like this case:

The supplier finally paid the credit note he made, we inserted a payment like this:
<img width="1072" alt="image" src="https://github.com/user-attachments/assets/99e4594b-13fd-4838-809a-a5acd4ac65a3" />

### Dolibarr Version

19

### Environment PHP

8.2

### Environment Database

MariaDB

**Comment by Cirillo007** [comment]
@eldy As for #32349 you mentioned that we cannot insert a negative amount in a supplier invoice, this bug is linked to this #32349

**Comment by eldy** [comment]
> @eldy As for #32349 you mentioned that we cannot insert a negative amount in a supplier invoice, this bug is linked to this #32349

Yes, looks similar trouble. The credit not did not have the type credit note so should not have been validated.
With the fix for v21, i hope this could not happen anymore.
